### PR TITLE
feat: Add process/service names to spans, fix ingester span propagation

### PR DESCRIPTION
**Pull request by alamb**
Closes https://github.com/influxdata/influxdb_iox/issues/7477

# Before this PR:
Not clear what the ingester spans are they appear disconnected:

Screenshot 2023-04-11 at 1 10 50 PM

# After this PR

After this PR it is clearer what IOx process/service the spans come from and also the ingester spans are correctly connected to their parents:

Screenshot 2023-04-11 at 1 01 40 PM

### CODE REVIEW

File: ingester2/src/buffer_tree/table.rs [hunk 1]
```diff
@@ -223,8 +223,9 @@ where
         );
 
         // Gather the partition data from all of the partitions in this table.
+        let span = SpanRecorder::new(span);
```
Comment (alamb) [inline ingester2/src/buffer_tree/table.rs:226]: To be honest I don't understand why the changes to use SpanRecorder rather than `span.map`  seems to make a difference connecting ingester spans but it repeatedly improved the display in Jaeger to include the spans under the "IOx ingester" section

**Review by philjb — APPROVED** [review]

**Comment by alamb** [comment]
Thanks @philjb ! I plan to leave this PR open until at least tomorrow in case anyone else would like to review it

File: ingester2/src/buffer_tree/table.rs (hunk 1 above)
Comment (crepererum) [reply ingester2/src/buffer_tree/table.rs:226]: because a `Span` does NOTHING on it's own. It's pure data. A recorder however emits span data. I agree that the trio Span/SpanCtx/SpanRecorder is not the best UX at the moment.

**Review by crepererum — APPROVED** [review]

**Review by domodwyer — APPROVED** [review]

File: ingester2/src/buffer_tree/table.rs (hunk 1 above)
Comment (alamb) [reply ingester2/src/buffer_tree/table.rs:226]: Thank you -- I tried to capture this sentiment into docs in https://github.com/influxdata/influxdb_iox/pull/7525
